src/runtime/virtcontainers/utils/utils.go

```diff
@@ -235,6 +235,17 @@ func BuildSocketPath(elements ...string) (string, error) {
 
 // SupportsVsocks returns true if vsocks are supported, otherwise false
 func SupportsVsocks() (bool, error) {
+	vsockModule := "vhost_vsock"
+	// First, check to see if the vhost_vsock module is already loaded
+	path := filepath.Join(sysModuleDir, vsockModule)
```

Comment:
I think this is prone to errors, what happen if the `vhost_vsock` is buit-in ?  you will try to load it anyway and will fail

Reply:
yes, if vhost_vsock built into kernel，modprobe  will be failed. How about to ignore cmd.CombinedOutput() returned value check?

Reply:
A couple of things:

- To check if a module is built-in, you can either look at `/lib/modules/$(uname -r)/modules.builtin`  or use `modprobe -D`:
  ```sh
  $ sudo modprobe -D ext4
  builtin ext4
  ```
- `kata-check` already does this sort of thing, so might be worth looking at it (it's slightly convoluted though I'm afraid :):
  https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/cli/kata-check_amd64.go#L36
  https://github.com/kata-containers/kata-containers/blob/2.0-dev/src/runtime/cli/kata-check.go#L133

Reply:
ok, thanks.